java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt

```diff
@@ -87,11 +88,18 @@ object Agent {
 
     private val pathSeparator = if (Platform.osFamily == OsFamily.WINDOWS) "\\" else "/"
 
+    private fun getDrillInstallationDir() = open("/proc/self/cmdline", O_RDONLY).let(::Input).readText().run {
+        val isContainsAgetPath: (String) -> Boolean = { it.contains("-agentpath:") }
+        val agentLine = this.takeIf(isContainsAgetPath) ?: getenv("JAVA_TOOL_OPTIONS")!!.toKString()
```

Comment:
In this case `.run` makes method code harder to comprehend
Consider:
```
fun getSomeValue() {
val source1 = ...
val source2 = ...
val result = if condition(source1) source1 else source2 
...
}
```
vs
```
fun getSomeValue() = source1.run {  // the whole lambda expression intuitively associates with the source 1
  val source2 = ...
  val result = this.takeIf(source1) ?: source2 // source 2 looks incidental
}
```

Reply:
Completely refactored and moved to agent lib.